aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/ReadHandler.java

```diff
@@ -28,25 +28,41 @@
 
         DescribeLogGroupsResponse response = null;
         ListTagsLogGroupResponse tagsResponse = null;
-        try {
-            response = proxy.injectCredentialsAndInvokeV2(Translator.translateToReadRequest(model),
-                    ClientBuilder.getClient()::describeLogGroups);
+        ResourceModel modelFromReadResult = null;
+        String nextToken = null;
+        // Keep paginating until requested log group is found
+        do {
             try {
-                tagsResponse = proxy.injectCredentialsAndInvokeV2(Translator.translateToListTagsLogGroupRequest(model.getLogGroupName()),
-                        ClientBuilder.getClient()::listTagsLogGroup);
-            } catch (final CloudWatchLogsException e) {
-                if (Translator.ACCESS_DENIED_ERROR_CODE.equals(e.awsErrorDetails().errorCode())) {
-                    // fail silently, if there is no permission to list tags
-                    logger.log(e.getMessage());
-                } else {
-                    throw e;
+                response = proxy.injectCredentialsAndInvokeV2(Translator.translateToReadRequest(model, nextToken),
+                        ClientBuilder.getClient()::describeLogGroups);
+                if (tagsResponse == null) {
+                    try {
+                        tagsResponse = proxy.injectCredentialsAndInvokeV2(Translator.translateToListTagsLogGroupRequest(model.getLogGroupName()),
```

Comment:
why are we calling listTags in the loop?  This is adding double the unnecessary API calls.  Shouldn't we just call it once after we found the right log group?

Reply:
I explained my thinking in this comment https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/pull/65#discussion_r717956209 

But you are right that we should just call it once we found the log group. I split up the implementation in new commit